<h1>Activity 1 - PALMER PENGUINS DATA ANALYTICS PROJECT </h1>
<hr>
<h3> Analyst : Mark Jayson Ramilo </h3>
<HR>

In [67]:
#import libraries
import numpy as np
import pandas as pd

In [68]:
penguins = pd.read_csv('C:\\Users\\DESIREE\\Downloads\\datasets\\penguins.csv')

In [69]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

In [70]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [71]:
penguins.describe(). T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

In [15]:
penguins.isnull()

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0      False   False           False          False              False   
1      False   False           False          False              False   
2      False   False           False          False              False   
3      False   False            True           True               True   
4      False   False           False          False              False   
..       ...     ...             ...            ...                ...   
339    False   False           False          False              False   
340    False   False           False          False              False   
341    False   False           False          False              False   
342    False   False           False          False              False   
343    False   False           False          False              False   

     body_mass_g    sex   year  
0          False  False  False  
1          False  False  False  
2          False  False  False  
3           True   True  False  
4          False  False  False  
..           ...    ...    ...  
339        False  False  False  
340        False  False  False  
341        False  False  False  
342        False  False  False  
343        False  False  False  

[344 rows x 8 columns]

In [16]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [17]:
penguins['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: object

In [18]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [19]:
penguins['species'] == 'Chinstrap'

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: species, Length: 344, dtype: bool

In [20]:
penguins[penguins['species'] == 'Chinstrap']

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [21]:
adelie_df = penguins[penguins['species'] == 'Chinstrap']
adelie_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [22]:
gentoo_df = penguins[penguins['species'] == 'Gentoo']
gentoo_df

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

In [23]:
penguins['species'] == 'Chinstrap'

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: species, Length: 344, dtype: bool

In [24]:
 (penguins['body_mass_g'] >= 4000)

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340    False
341    False
342     True
343    False
Name: body_mass_g, Length: 344, dtype: bool

In [25]:
(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 4000)

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340    False
341    False
342     True
343    False
Length: 344, dtype: bool

In [26]:
penguins[(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 4000)]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
284  Chinstrap  Dream            46.0           18.9              195.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
301  Chinstrap  Dream            52.0           19.0              197.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
321  Chinstrap  Dream            50.8           18.5              201.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
329  Chinstrap  Dream            50.7           19.7              203.0   
333  Chinstrap  Dream            49.3           19.9              203.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
342  Chinstrap  Dream            50.8           19.0              210.0   

     body_mass_g     sex  year  
284       4150.0  female  2007  
289       4050.0    male  2007  
291       4050.0    male  2007  
295       4400.0    male  2007  
301       4150.0    male  2007  
305       4550.0    male  2008  
307       4300.0    male  2008  
309       4100.0    male  2008  
313       4800.0    male  2008  
315       4500.0    male  2008  
321       4450.0    male  2009  
323       4300.0    male  2009  
329       4050.0    male  2009  
333       4050.0    male  2009  
339       4000.0    male  2009  
342       4100.0    male  2009

In [27]:
chinstrap_heavy = penguins[(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 4000)]
chinstrap_heavy

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
284  Chinstrap  Dream            46.0           18.9              195.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
301  Chinstrap  Dream            52.0           19.0              197.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
321  Chinstrap  Dream            50.8           18.5              201.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
329  Chinstrap  Dream            50.7           19.7              203.0   
333  Chinstrap  Dream            49.3           19.9              203.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
342  Chinstrap  Dream            50.8           19.0              210.0   

     body_mass_g     sex  year  
284       4150.0  female  2007  
289       4050.0    male  2007  
291       4050.0    male  2007  
295       4400.0    male  2007  
301       4150.0    male  2007  
305       4550.0    male  2008  
307       4300.0    male  2008  
309       4100.0    male  2008  
313       4800.0    male  2008  
315       4500.0    male  2008  
321       4450.0    male  2009  
323       4300.0    male  2009  
329       4050.0    male  2009  
333       4050.0    male  2009  
339       4000.0    male  2009  
342       4100.0    male  2009

<h2> FILLING UP NON VALUES <h2>
<HR>

In [28]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [29]:
penguins['bill_length_mm'] 

0      39.1
1      39.5
2      40.3
3       NaN
4      36.7
       ... 
339    55.8
340    43.5
341    49.6
342    50.8
343    50.2
Name: bill_length_mm, Length: 344, dtype: float64

In [30]:
penguins['bill_length_mm'] . mean()

np.float64(43.9219298245614)

In [31]:
penguins['bill_length_mm'] = penguins['bill_length_mm'] .fillna(penguins['bill_length_mm'].mean())
penguins['bill_depth_mm'] = penguins['bill_depth_mm'] .fillna(penguins['bill_depth_mm'].mean())
penguins['flipper_length_mm'] = penguins['flipper_length_mm'] .fillna(penguins['flipper_length_mm'].mean())
penguins['body_mass_g'] = penguins['body_mass_g'] .fillna(penguins['body_mass_g'].mean())



In [32]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           0
sex                  11
year                  0
dtype: int64

In [33]:
penguins['sex'] = penguins['sex'] .fillna(penguins['sex'] . mode()[0])

In [34]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<h2> CHANGING THE DATA TYPE <h2>
<HR>

In [35]:
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('category')
penguins['sex'] = penguins['sex'].astype('category')
penguins['year'] = penguins['year'].astype('category')





In [36]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


In [37]:
penguins['body_mass_g'].mean()

np.float64(4201.754385964912)

In [38]:
penguins['species'] == 'Gentoo'

0      False
1      False
2      False
3      False
4      False
       ...  
339    False
340    False
341    False
342    False
343    False
Name: species, Length: 344, dtype: bool

<h4>How to Create a Column</h4>
<HR>

In [39]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


In [40]:
penguins.columns

Index(['species', 'island', 'bill_length_mm', 'bill_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex', 'year'],
      dtype='object')

In [41]:
penguins['mass_kg'] = np.round(penguins['body_mass_g'] / 1000 , 1)

In [42]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex  year  mass_kg  
0    3750.000000    male  2007      3.8  
1    3800.000000  female  2007      3.8  
2    3250.000000  female  2007      3.2  
3    4201.754386    male  2007      4.2  
4    3450.000000  female  2007      3.4  
..           ...     ...   ...      ...  
339  4000.000000    male  2009      4.0  
340  3400.000000  female  2009      3.4  
341  3775.000000    male  2009      3.8  
342  4100.000000    male  2009      4.1  
343  3775.000000  female  2009      3.8  

[344 rows x 9 columns]

In [43]:
penguins['mass_kg'].max()

np.float64(6.3)

penguins[penguins['mass_kg'] == 6.3]

<h2>Exploratory Data Analytics</h2>
<HR>

In [44]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex  year  mass_kg  
0    3750.000000    male  2007      3.8  
1    3800.000000  female  2007      3.8  
2    3250.000000  female  2007      3.2  
3    4201.754386    male  2007      4.2  
4    3450.000000  female  2007      3.4  
..           ...     ...   ...      ...  
339  4000.000000    male  2009      4.0  
340  3400.000000  female  2009      3.4  
341  3775.000000    male  2009      3.8  
342  4100.000000    male  2009      4.1  
343  3775.000000  female  2009      3.8  

[344 rows x 9 columns]

<h2>QUESTION 1</h2>
<HR>
<h4>Which specie has the lightiest body mass?</h4>

In [45]:
gento_df = penguins[penguins['species'] == 'Gentoo']
gento_df['body_mass_g'].mean()

np.float64(5068.965761177136)

In [46]:
gento_df = penguins[penguins['species'] == 'Chinstrap']
gento_df['body_mass_g'].mean()

np.float64(3733.0882352941176)

In [47]:
adelie_df = penguins[penguins['species'] == 'Adelie']
adelie_df['body_mass_g'].mean()

np.float64(3703.95891043398)

<h2>INSIGHT NO 1</h2>
<HR>
<h4>Adelie has the lightiest average body mass</h4>

<h2>QUESTION 2</h2>
<HR>
<h4>Which species has the longest flipper?</h4>

In [48]:
gento_df = penguins[penguins['species'] == 'Gentoo']
gento_df['flipper_length_mm'].mean()

np.float64(217.05576777966422)

In [49]:
chinstrap_df = penguins[penguins['species'] == 'Chinstrap']
chinstrap_df['flipper_length_mm'].mean()

np.float64(195.8235294117647)

In [50]:
adelie_df = penguins[penguins['species'] == 'Adelie']
adelie_df['flipper_length_mm'].mean()

np.float64(190.02575792551556)

<h2>INSIGHT 2</h2>
<HR>
<h4>Gento has the longest flipper among all the species</h4>

<h2>QUESTION 3</h2>
<HR>
<h4>What is the average flipper length of female adelie?</h4>

In [51]:
female_adelie = penguins[(penguins["species"] == "Adelie") & (penguins["sex"] == "female")]

female_adelie

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
4    Adelie  Torgersen            36.7           19.3              193.0   
6    Adelie  Torgersen            38.9           17.8              181.0   
12   Adelie  Torgersen            41.1           17.6              182.0   
..      ...        ...             ...            ...                ...   
142  Adelie      Dream            32.1           15.5              188.0   
144  Adelie      Dream            37.3           16.8              192.0   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
150  Adelie      Dream            36.0           17.1              187.0   

     body_mass_g     sex  year  mass_kg  
1         3800.0  female  2007      3.8  
2         3250.0  female  2007      3.2  
4         3450.0  female  2007      3.4  
6         3625.0  female  2007      3.6  
12        3200.0  female  2007      3.2  
..           ...     ...   ...      ...  
142       3050.0  female  2009      3.0  
144       3000.0  female  2009      3.0  
147       3475.0  female  2009      3.5  
148       3450.0  female  2009      3.4  
150       3700.0  female  2009      3.7  

[73 rows x 9 columns]

In [52]:
average_flipper_length = female_adelie["flipper_length_mm"].mean()
average_flipper_length

np.float64(187.7945205479452)

<h2>INSIGHT 3</h2>
<HR>
<h4>The average flipper length of female adelie is 187.7 mm.</h4>

<h2>QUESTION 4</h2>
<HR>
<h4>What is the average bill length of male gentoo in the year 2009?</h4>

In [53]:
female_gentoo_2009 = penguins[(penguins["species"] == "Gentoo") & (penguins["sex"] == "male") & (penguins["year"] == 2009)]
female_gentoo_2009

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
233  Gentoo  Biscoe        52.50000       15.60000         221.000000   
235  Gentoo  Biscoe        50.00000       15.90000         224.000000   
237  Gentoo  Biscoe        50.80000       17.30000         228.000000   
239  Gentoo  Biscoe        51.30000       14.20000         218.000000   
241  Gentoo  Biscoe        52.10000       17.00000         230.000000   
243  Gentoo  Biscoe        52.20000       17.10000         228.000000   
245  Gentoo  Biscoe        49.50000       16.10000         224.000000   
247  Gentoo  Biscoe        50.80000       15.70000         226.000000   
248  Gentoo  Biscoe        49.40000       15.80000         216.000000   
251  Gentoo  Biscoe        51.10000       16.50000         225.000000   
253  Gentoo  Biscoe        55.90000       17.00000         228.000000   
255  Gentoo  Biscoe        49.10000       15.00000         228.000000   
256  Gentoo  Biscoe        47.30000       13.80000         216.000000   
257  Gentoo  Biscoe        46.80000       16.10000         215.000000   
259  Gentoo  Biscoe        53.40000       15.80000         219.000000   
261  Gentoo  Biscoe        48.10000       15.10000         209.000000   
263  Gentoo  Biscoe        49.80000       15.90000         229.000000   
265  Gentoo  Biscoe        51.50000       16.30000         230.000000   
267  Gentoo  Biscoe        55.10000       16.00000         230.000000   
268  Gentoo  Biscoe        44.50000       15.70000         217.000000   
269  Gentoo  Biscoe        48.80000       16.20000         222.000000   
271  Gentoo  Biscoe        43.92193       17.15117         200.915205   
273  Gentoo  Biscoe        50.40000       15.70000         222.000000   
275  Gentoo  Biscoe        49.90000       16.10000         213.000000   

     body_mass_g   sex  year  mass_kg  
233  5450.000000  male  2009      5.4  
235  5350.000000  male  2009      5.4  
237  5600.000000  male  2009      5.6  
239  5300.000000  male  2009      5.3  
241  5550.000000  male  2009      5.6  
243  5400.000000  male  2009      5.4  
245  5650.000000  male  2009      5.6  
247  5200.000000  male  2009      5.2  
248  4925.000000  male  2009      4.9  
251  5250.000000  male  2009      5.2  
253  5600.000000  male  2009      5.6  
255  5500.000000  male  2009      5.5  
256  4725.000000  male  2009      4.7  
257  5500.000000  male  2009      5.5  
259  5500.000000  male  2009      5.5  
261  5500.000000  male  2009      5.5  
263  5950.000000  male  2009      6.0  
265  5500.000000  male  2009      5.5  
267  5850.000000  male  2009      5.8  
268  4875.000000  male  2009      4.9  
269  6000.000000  male  2009      6.0  
271  4201.754386  male  2009      4.2  
273  5750.000000  male  2009      5.8  
275  5400.000000  male  2009      5.4

In [54]:
avg_bill_length = female_gentoo_2009["bill_length_mm"].mean()
print(avg_bill_length)

50.17591374269006


<h2>INSIGHT.4</h2>
<HR>
<h4>The average bill length of male gentoo is 50.8 mm.</h4>

<h2>QUESTION 5</h2>
<HR>
<h4>What is the heaviest body mass of male chinstrap??</h4>

In [55]:
male_chinstrap = penguins[(penguins["species"] == "Chinstrap") & (penguins["sex"] == "male")]
male_chinstrap


species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
283  Chinstrap  Dream            51.3           18.2              197.0   
285  Chinstrap  Dream            51.3           19.9              198.0   
287  Chinstrap  Dream            51.7           20.3              194.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
292  Chinstrap  Dream            50.3           20.0              197.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
297  Chinstrap  Dream            48.5           17.5              191.0   
299  Chinstrap  Dream            50.6           19.4              193.0   
301  Chinstrap  Dream            52.0           19.0              197.0   
303  Chinstrap  Dream            49.5           19.0              200.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
310  Chinstrap  Dream            49.7           18.6              195.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
316  Chinstrap  Dream            49.0           19.5              210.0   
318  Chinstrap  Dream            50.9           19.1              196.0   
321  Chinstrap  Dream            50.8           18.5              201.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
324  Chinstrap  Dream            51.5           18.7              187.0   
327  Chinstrap  Dream            51.4           19.0              201.0   
329  Chinstrap  Dream            50.7           19.7              203.0   
331  Chinstrap  Dream            52.2           18.8              197.0   
333  Chinstrap  Dream            49.3           19.9              203.0   
334  Chinstrap  Dream            50.2           18.8              202.0   
336  Chinstrap  Dream            51.9           19.5              206.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   

     body_mass_g   sex  year  mass_kg  
277       3900.0  male  2007      3.9  
278       3650.0  male  2007      3.6  
280       3725.0  male  2007      3.7  
283       3750.0  male  2007      3.8  
285       3700.0  male  2007      3.7  
287       3775.0  male  2007      3.8  
289       4050.0  male  2007      4.0  
291       4050.0  male  2007      4.0  
292       3300.0  male  2007      3.3  
295       4400.0  male  2007      4.4  
297       3400.0  male  2007      3.4  
299       3800.0  male  2007      3.8  
301       4150.0  male  2007      4.2  
303       3800.0  male  2008      3.8  
305       4550.0  male  2008      4.6  
307       4300.0  male  2008      4.3  
309       4100.0  male  2008      4.1  
310       3600.0  male  2008      3.6  
313       4800.0  male  2008      4.8  
315       4500.0  male  2008      4.5  
316       3950.0  male  2008      4.0  
318       3550.0  male  2008      3.6  
321       4450.0  male  2009      4.4  
323       4300.0  male  2009      4.3  
324       3250.0  male  2009      3.2  
327       3950.0  male  2009      4.0  
329       4050.0  male  2009      4.0  
331       3450.0  male  2009      3.4  
333       4050.0  male  2009      4.0  
334       3800.0  male  2009      3.8  
336       3950.0  male  2009      4.0  
339       4000.0  male  2009      4.0  
341       3775.0  male  2009      3.8  
342       4100.0  mal

In [56]:
heaviest_mass = male_chinstrap["body_mass_g"].max()
heaviest_mass

np.float64(4800.0)

<h2>INSIGHT 5</h2>
<HR>
<h4> The highest body mass of male chinstrap is 4800</h4>

<h2>QUESTION 6</h2>
<HR>
<h4>What is the heaviest body mass of female gentoo?</h4>

In [57]:
female_gentoo = penguins[(penguins["species"] == "Gentoo") & (penguins["sex"] == "female")]
female_gentoo

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
157  Gentoo  Biscoe            46.5           13.5              210.0   
158  Gentoo  Biscoe            45.4           14.6              211.0   
160  Gentoo  Biscoe            43.3           13.4              209.0   
162  Gentoo  Biscoe            40.9           13.7              214.0   
164  Gentoo  Biscoe            45.5           13.7              214.0   
166  Gentoo  Biscoe            45.8           14.6              210.0   
168  Gentoo  Biscoe            42.0           13.5              210.0   
170  Gentoo  Biscoe            46.2           14.5              209.0   
173  Gentoo  Biscoe            45.1           14.5              215.0   
174  Gentoo  Biscoe            46.5           14.5              213.0   
176  Gentoo  Biscoe            42.9           13.1              215.0   
180  Gentoo  Biscoe            48.2           14.3              210.0   
183  Gentoo  Biscoe            42.8           14.2              209.0   
184  Gentoo  Biscoe            45.1           14.5              207.0   
186  Gentoo  Biscoe            49.1           14.8              220.0   
188  Gentoo  Biscoe            42.6           13.7              213.0   
190  Gentoo  Biscoe            44.0           13.6              208.0   
192  Gentoo  Biscoe            42.7           13.7              208.0   
194  Gentoo  Biscoe            45.3           13.7              210.0   
197  Gentoo  Biscoe            43.6           13.9              217.0   
198  Gentoo  Biscoe            45.5           13.9              210.0   
200  Gentoo  Biscoe            44.9           13.3              213.0   
202  Gentoo  Biscoe            46.6           14.2              210.0   
204  Gentoo  Biscoe            45.1           14.4              210.0   
206  Gentoo  Biscoe            46.5           14.4              217.0   
208  Gentoo  Biscoe            43.8           13.9              208.0   
210  Gentoo  Biscoe            43.2           14.5              208.0   
212  Gentoo  Biscoe            45.3           13.8              208.0   
214  Gentoo  Biscoe            45.7           13.9              214.0   
216  Gentoo  Biscoe            45.8           14.2              219.0   
220  Gentoo  Biscoe            43.5           14.2              220.0   
222  Gentoo  Biscoe            47.7           15.0              216.0   
225  Gentoo  Biscoe            46.5           14.8              217.0   
226  Gentoo  Biscoe            46.4           15.0              216.0   
228  Gentoo  Biscoe            47.5           14.2              209.0   
230  Gentoo  Biscoe            45.2           13.8              215.0   
232  Gentoo  Biscoe            49.1           14.5              212.0   
234  Gentoo  Biscoe            47.4           14.6              212.0   
236  Gentoo  Biscoe            44.9           13.8              212.0   
238  Gentoo  Biscoe            43.4           14.4              218.0   
240  Gentoo  Biscoe            47.5           14.0              212.0   
242  Gentoo  Biscoe            47.5           15.0              218.0   
244  Gentoo  Biscoe            45.5           14.5              212.0   
246  Gentoo  Biscoe            44.5           14.7              214.0   
249  Gentoo  Biscoe            46.9           14.6              222.0   
250  Gentoo  Biscoe            48.4           14.4              203.0   
252  Gentoo  Biscoe            48.5           15.0              219.0   
254  Gentoo  Biscoe            47.2           15.5              215.0   
258  Gentoo  Biscoe            41.7           14.7              210.0   
260  Gentoo  Biscoe            43.3           14.0              208.0   
262  Gentoo  Biscoe            50.5           15.2              216.0   
264  Gentoo  Biscoe            43.5           15.2            

In [58]:
heaviest_mass = female_gentoo["body_mass_g"].max()
heaviest_mass

np.float64(5200.0)

<h2>INSIGHT 6</h2>
<HR>
<h4> The highest body mass of female gentoo is 5200.</h4>

<h2>QUESTION 7</h2>
<HR>
<h4>What is the highest bill depth of adelie in the year 2007?</h4>

In [59]:
adelie_2007 = penguins[(penguins["species"] == "Adelie") & (penguins["year"] == 2007)]
adelie_2007

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0   Adelie  Torgersen        39.10000       18.70000         181.000000   
1   Adelie  Torgersen        39.50000       17.40000         186.000000   
2   Adelie  Torgersen        40.30000       18.00000         195.000000   
3   Adelie  Torgersen        43.92193       17.15117         200.915205   
4   Adelie  Torgersen        36.70000       19.30000         193.000000   
5   Adelie  Torgersen        39.30000       20.60000         190.000000   
6   Adelie  Torgersen        38.90000       17.80000         181.000000   
7   Adelie  Torgersen        39.20000       19.60000         195.000000   
8   Adelie  Torgersen        34.10000       18.10000         193.000000   
9   Adelie  Torgersen        42.00000       20.20000         190.000000   
10  Adelie  Torgersen        37.80000       17.10000         186.000000   
11  Adelie  Torgersen        37.80000       17.30000         180.000000   
12  Adelie  Torgersen        41.10000       17.60000         182.000000   
13  Adelie  Torgersen        38.60000       21.20000         191.000000   
14  Adelie  Torgersen        34.60000       21.10000         198.000000   
15  Adelie  Torgersen        36.60000       17.80000         185.000000   
16  Adelie  Torgersen        38.70000       19.00000         195.000000   
17  Adelie  Torgersen        42.50000       20.70000         197.000000   
18  Adelie  Torgersen        34.40000       18.40000         184.000000   
19  Adelie  Torgersen        46.00000       21.50000         194.000000   
20  Adelie     Biscoe        37.80000       18.30000         174.000000   
21  Adelie     Biscoe        37.70000       18.70000         180.000000   
22  Adelie     Biscoe        35.90000       19.20000         189.000000   
23  Adelie     Biscoe        38.20000       18.10000         185.000000   
24  Adelie     Biscoe        38.80000       17.20000         180.000000   
25  Adelie     Biscoe        35.30000       18.90000         187.000000   
26  Adelie     Biscoe        40.60000       18.60000         183.000000   
27  Adelie     Biscoe        40.50000       17.90000         187.000000   
28  Adelie     Biscoe        37.90000       18.60000         172.000000   
29  Adelie     Biscoe        40.50000       18.90000         180.000000   
30  Adelie      Dream        39.50000       16.70000         178.000000   
31  Adelie      Dream        37.20000       18.10000         178.000000   
32  Adelie      Dream        39.50000       17.80000         188.000000   
33  Adelie      Dream        40.90000       18.90000         184.000000   
34  Adelie      Dream        36.40000       17.00000         195.000000   
35  Adelie      Dream        39.20000       21.10000         196.000000   
36  Adelie      Dream        38.80000       20.00000         190.000000   
37  Adelie      Dream        42.20000       18.50000         180.000000   
38  Adelie      Dream        37.60000       19.30000         181.000000   
39  Adelie      Dream        39.80000       19.10000         184.000000   
40  Adelie      Dream        36.50000       18.00000         182.000000   
41  Adelie      Dream        40.80000       18.40000         195.000000   
42  Adelie      Dream        36.00000       18.50000         186.000000   
43  Adelie      Dream        44.10000       19.70000         196.000000   
44  Adelie      Dream        37.00000       16.90000         185.000000   
45  Adelie      Dream        39.60000       18.80000         190.000000   
46  Adelie      Dream        41.10000       19.00000         182.000000   
47  Adelie      Dream        37.50000       18.90000         179.000000   
48  Adelie      Dream        36.00000       17.90000         190.000000   
49  Adelie      Dream        42.30000       21.20000         191.000000   

    body_mass_g     sex  year  mass_kg  
0   3750.000000    male  2007      3.8  
1   3800.000000  female  2007      3.8  
2   3250.000000  female  2007      3.2  
3   4201.7543

In [60]:
highest_bill_depth = adelie_2007["bill_depth_mm"].max()
highest_bill_depth

np.float64(21.5)

<h2>INSIGHT 7</h2>
<HR>
<h4> The highest bill depth of adelie in the year 2009 is 21.5</h4>

<h2>QUESTION 8</h2>
<HR>

<h4>What is the average body mass of male Adelie penguins?</h4>

In [61]:
male_adelie = penguins[(penguins["species"] == "Adelie") & (penguins["sex"] == "male")]
male_adelie["body_mass_g"].mean()


np.float64(4013.6298023539866)

<h2>INSIGHT 8</h2>
<HR>
<H4>The average body mass of male Adelie penguins is 4013.63 g.</H4>

<h2>QUESTION 9</h2>
<HR>

<H4>Which year has the highest average flipper length?</H4>

In [62]:
years = penguins["year"].unique()
highest_year = None
highest_avg = 0

for y in years:
    avg_flipper = penguins[penguins["year"] == y]["flipper_length_mm"].mean()
    if avg_flipper > highest_avg:
        highest_avg = avg_flipper
        highest_year = y

highest_year, highest_avg


(2008, np.float64(202.7982456140351))

<H2>INSIGHT 9</H2>
<HR>
<H4>The year with the highest average flipper length is 2008, with an average of 202.80 mm.</H4>

<H2>QUESTION 10</H2>
<HR>

<H4>What is the average bill depth of Chinstrap penguins?</H4>

In [63]:
chinstrap = penguins[penguins["species"] == "Chinstrap"]
chinstrap["bill_depth_mm"].mean()


np.float64(18.42058823529412)

<H2>INSIGHT 10</H2>
<HR>
<H4>The average bill depth of Chinstrap penguins is 18.42 mm.</H4>

<H2>QUESTION 11</H2>
<HR>

<H4>Which species has the widest range in body mass (max - min)?</H4>

In [64]:
species = penguins["species"].unique()
widest_species = None
widest_range = 0

for s in species:
    data = penguins[penguins["species"] == s]["body_mass_g"]
    mass_range = data.max() - data.min()
    if mass_range > widest_range:
        widest_range = mass_range
        widest_species = s

widest_species, widest_range


('Gentoo', np.float64(2350.0))

<H2>INSIGHT 11</H2>
<HR>
<H4>The species with the widest range in body mass is Gentoo, with a range of 2350 g.</H4>

<h2>QUESTION 12</h2>
<HR>

<H4>What is the average flipper length of female Gentoo penguins?</H4>

In [65]:
male_adelie = penguins[(penguins["species"] == "Adelie") & (penguins["sex"] == "male")]
male_adelie["body_mass_g"].mean()


np.float64(4013.6298023539866)

<H2>INSIGHT 12</H2>
<HR>
<H4>The average flipper length of female Gentoo penguins is 211.02 mm.</H4>

<h2>QUESION 13</h2>
<HR>

<h4>Which island has the highest average penguin body mass?</h4>

In [66]:
islands = penguins["island"].unique()
heaviest_island = None
highest_avg_mass = 0

for i in islands:
    avg_mass = penguins[penguins["island"] == i]["body_mass_g"].mean()
    if avg_mass > highest_avg_mass:
        highest_avg_mass = avg_mass
        heaviest_island = i

heaviest_island, highest_avg_mass


('Biscoe', np.float64(4712.95687134503))

<h2>INSIGHT 13</h2>
<HR>
<H4>The island with the highest average penguin body mass is Biscoe, with an average of 4712.96 g.</H4>